Userland/Libraries/LibGfx/PBMLoader.cpp

```diff
@@ -7,19 +7,21 @@
 
 #include "PBMLoader.h"
 #include "PortableImageLoaderCommon.h"
-#include "Streamer.h"
-#include <AK/Endian.h>
-#include <string.h>
 
 namespace Gfx {
 
-bool read_image_data(PBMLoadingContext& context, Streamer& streamer)
+ErrorOr<void> read_image_data(PBMLoadingContext& context)
 {
-    u8 byte;
+    auto& stream = *context.stream;
     Vector<Gfx::Color> color_data;
 
     if (context.type == PBMLoadingContext::Type::ASCII) {
-        while (streamer.read(byte)) {
+        while (true) {
```

Comment:
Maybe `while (!stream.is_eof()) {` and `auto byte = TRY(stream.read_value<u8>())`?

(Any time you manually check `is_error()`, that's a code smell imho.)

Reply:
> Any time you manually check is_error(), that's a code smell imho.

I agree with that, but I'm not sure that `while (!stream.is_eof())` is a good solution either.
It only works because it's a `MemoryStream` which does not conform to posix's "read return 0 once before the actual eof" which isn't particularly good as we might want to directly read from a file in the future.